internal/query/projection/project_metadata.go

```diff
@@ -0,0 +1,168 @@
+package projection
+
+import (
+	"context"
+
+	"github.com/zitadel/zitadel/internal/eventstore"
+	old_handler "github.com/zitadel/zitadel/internal/eventstore/handler"
+	"github.com/zitadel/zitadel/internal/eventstore/handler/v2"
+	"github.com/zitadel/zitadel/internal/repository/instance"
+	"github.com/zitadel/zitadel/internal/repository/org"
+	"github.com/zitadel/zitadel/internal/repository/project"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+const (
+	ProjectMetadataProjectionTable = "projections.project_metadata"
+
+	ProjectMetadataColumnProjectID     = "project_id"
+	ProjectMetadataColumnCreationDate  = "creation_date"
+	ProjectMetadataColumnChangeDate    = "change_date"
+	ProjectMetadataColumnSequence      = "sequence"
+	ProjectMetadataColumnResourceOwner = "resource_owner"
+	ProjectMetadataColumnInstanceID    = "instance_id"
+	ProjectMetadataColumnKey           = "key"
+	ProjectMetadataColumnValue         = "value"
+)
+
+type projectMetadataProjection struct{}
+
+func (*projectMetadataProjection) Name() string {
+	return ProjectMetadataProjectionTable
+}
+
+func (*projectMetadataProjection) Init() *old_handler.Check {
+	return handler.NewTableCheck(
+		handler.NewTable([]*handler.InitColumn{
+			handler.NewColumn(ProjectMetadataColumnProjectID, handler.ColumnTypeText),
+			handler.NewColumn(ProjectMetadataColumnCreationDate, handler.ColumnTypeTimestamp),
+			handler.NewColumn(ProjectMetadataColumnChangeDate, handler.ColumnTypeTimestamp),
+			handler.NewColumn(ProjectMetadataColumnSequence, handler.ColumnTypeInt64),
+			handler.NewColumn(ProjectMetadataColumnResourceOwner, handler.ColumnTypeText),
+			handler.NewColumn(ProjectMetadataColumnInstanceID, handler.ColumnTypeText),
+			handler.NewColumn(ProjectMetadataColumnKey, handler.ColumnTypeText),
+			handler.NewColumn(ProjectMetadataColumnValue, handler.ColumnTypeBytes, handler.Nullable()),
+		},
+			handler.NewPrimaryKey(ProjectMetadataColumnInstanceID, ProjectMetadataColumnProjectID, ProjectMetadataColumnKey),
+			handler.WithIndex(handler.NewIndex(ProjectMetadataColumnValue, []string{ProjectMetadataColumnValue})),
+		),
+	)
+}
+
+func newProjectMetadataProjection(ctx context.Context, config handler.Config) *handler.Handler {
+	return handler.NewHandler(ctx, &config, new(projectMetadataProjection))
+}
+
+func (p *projectMetadataProjection) Reducers() []handler.AggregateReducer {
+	return []handler.AggregateReducer{
+		{
+			Aggregate: project.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  project.MetadataSetType,
```

Comment:
Please also add the `project.ProjectRemovedType`.

Reply:
Done